code/modules/hydroponics/seed_extractor.dm

```diff
@@ -58,6 +60,24 @@
 	. = ..()
 	register_context()
 
+/obj/machinery/seed_extractor/update_icon_state()
+	if(machine_stat & BROKEN)
+		if (!powered())
+			if (state_open)
+				icon_state = "[initial(icon_state)]-broken-off-open"
+			else
+				icon_state = "[initial(icon_state)]-broken-off"
+		else if (state_open)
+			icon_state = "[initial(icon_state)]-broken-open"
+		else
+			icon_state = "[initial(icon_state)]-broken"
```

Comment:
I think you could format this to make it a bit easier to expand / understand: 
```dm
/obj/machinery/seed_extractor/update_icon_state()
	icon_state = base_icon_state
	if((machine_stat & BROKEN)
		 icon_state += "-broken"
	if(!powered())
		icon_state += "-off"
	if(state_open)
		icon_state += "-open"
	return ..()
```

Reply:
You should probably also ensure icon state is updating whenever these changes, such as
```
/obj/machinery/seed_extractor/on_set_is_operational(old_value)
	update_appearance(UPDATE_ICON_STATE)
```